src/lib/components/Layer/Layer.js

```diff
@@ -3,45 +3,96 @@ import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
 import { Actions, prepareLayer, lngLat } from 'gisida';
 
-
 const mapStateToProps = (state, ownProps) => {
   const { APP, LOC } = state;
-  const MAP = state[ownProps.mapId]
+  const MAP = state[ownProps.mapId];
   if (!!!MAP) {
-    throw "MAP not found"
+    throw 'MAP not found';
   }
   return {
     APP,
     LOC,
     timeSeriesObj: MAP.timeseries[MAP.visibleLayerId],
     timeseries: MAP.timeseries,
-    layers: MAP.layers
-  }
-}
+    layers: MAP.layers,
+    primaryLayer: MAP.primaryLayer,
+    activeLayerIds: MAP.activeLayerIds,
+  };
+};
 
 export class Layer extends Component {
-
-  onLayerToggle = layer => {
+  onLayerToggle = (layer, e) => {
     // dispatch toggle layer action
     const { mapId, APP, LOC } = this.props;
     if (!mapId) {
       return null;
     }
+
+    const layerId = layer.id.replace('.json', '');
+    let pageURL = `${window.location.href}`;
+    /**
+     * Check for visibility. If false it means layer has been selected push to layer id to url
+     * else if visible it means layer has been checked off pop layer from url
+     */
+    if (layer && layer.visible === false) {
+      if (!window.location.href.includes('?layers=') && !window.location.href.includes('&primary=')) {
+        /**
+         * Query param `layers` does not exist. Have the layerId as the first
+         * value of query param `layers`. The assumption made is that
+         * there exists no other query params
+         */
+        pageURL = `${pageURL}?layers=${layerId}&primary=${layerId}`;
+      } else {
+        /**
+         * Query param `layers` exists. Add to exist list
+         * Update primary layer accordingly
+         */
+        pageURL = `${pageURL.split('&')[0]},${layerId}&primary=${layerId}`;
+        // pageURL.splice pageURL.indexOf('&primary');
+      }
+
+      history.pushState('', '', pageURL);
```

Comment:
@kahummer @kelvin-muchiri it would be really cool if these booleans and strings could be pulled into utility getters

Reply:
eg:
```js
import { 
  getUrlDoesHaveLayers,
  urlDoesHavePrimaryLayer,
  getNewUrl,
  addLayerToUrl,
  removeLayerFromUrl,
  setPrimaryLayer,
  ...
} from './some/new/helper/module';

...

const urlDoesHaveLayers = getUrlDoesHaveLayers(...)
const urlDoesHavePrimaryLayer = getDoesUrlHaveLayers(...)

if (!urlDoesHaveLayers && !urlDoesHavePrimaryLayer) {
  pageURL = getNewUrl(layerId)
} else {
  pageURL = addLayerToUrl(layerId)
}

...
```